tests/test_utils.py

```diff
@@ -274,20 +281,24 @@ def test_standard_zip_dir_with_env_variable(
             os.path.join("subdir", os.path.basename(file_in_a_subdir))
             not in zip_file.namelist()
         )
-        assert os.path.join(self.FILE_IN_SECOND_TEST_DIRECTORY) in zip_file.namelist()
+        assert Path(file_in_other_directory).name in zip_file.namelist()
 
-    def test_get_snowflake_packages(self, streamlit_requirements_txt):
-        os.chdir(".tests")
+    def test_get_snowflake_packages(
+        self, temp_test_directory: str, streamlit_requirements_txt
+    ):
+        os.chdir(temp_test_directory)
+        os.rename(streamlit_requirements_txt, self.REQUIREMENTS_SNOWFLAKE)
         result = utils.get_snowflake_packages()
-        os.chdir("..")
 
         assert result == test_data.requirements
 
-    def test_get_snowflake_packages_delta(self, streamlit_requirements_txt):
+    def test_get_snowflake_packages_delta(
+        self, temp_test_directory: str, streamlit_requirements_txt
+    ):
         anaconda_package = test_data.requirements[-1]
-        os.chdir(".tests")
+        os.chdir(temp_test_directory)
```

Comment:
Is it safe to stay in a temp dir after a test?

The same question to other usages of `chdir`.

Reply:
Looks good, but I would wrap it in a fixture.

I have such util in my PR with integration tests (https://github.com/Snowflake-Labs/snowcli/pull/245/files#diff-9ef616b95f50c54f5ca3b20f74dab7bdb579c6abd97a1c9cb8d624449bb6da8dR20).

I don't know if we want to share utils between tests and integration tests, so if you want to, you may simply copy-paste it. WDYT?

```python
class WorkingDirectoryChanger:
    def __init__(self):
        self._initial_working_directory = os.getcwd()

    @staticmethod
    def change_working_directory_to(directory: str):
        os.chdir(directory)

    def restore_initial_working_directory(self):
        self.change_working_directory_to(self._initial_working_directory)


@pytest.fixture
def temporary_working_directory():
    working_directory_changer = WorkingDirectoryChanger()
    with TemporaryDirectory() as tmp_dir:
        working_directory_changer.change_working_directory_to(tmp_dir)
        yield tmp_dir
        working_directory_changer.restore_initial_working_directory()

```

Reply:
I tried few different approaches (yours, pure fixture, decorator), and i chose using separate fixture. It`s not as versatile as your Changer, but i only need some tests to enter temp directory, and this way i avoid duplicating large chunks of code